Mailer/app/api/v1/Handlers/Mailers/MailLogsHandler.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Remp\MailerModule\Api\v1\Handlers\Mailers;
+
+use Nette\Utils\Arrays;
+use Nette\Utils\Json;
+use Remp\MailerModule\ActiveRow;
+use Remp\MailerModule\Api\Validation\InputValidatorTrait;
+use Remp\MailerModule\Repository\LogsRepository;
+use Tomaj\NetteApi\Handlers\BaseHandler;
+use Tomaj\NetteApi\Response\JsonApiResponse;
+
+class MailLogsHandler extends BaseHandler
+{
+    private $logsRepository;
+
+    use InputValidatorTrait;
+
+    public function __construct(
+        LogsRepository $logsRepository
+    ) {
+        parent::__construct();
+        $this->logsRepository = $logsRepository;
+    }
+
+    public function handle($payload)
+    {
+        $request = file_get_contents("php://input");
+        $payload = $this->validateInput($request, 'logs.schema.json');
+
+        if ($this->hasErrors()){
+            return $payload;
+        }
+
+        $output = $this->logsRepository->all()->where($payload['filter'] . ' IS NOT NULL');
+
+        if (!empty($payload['email'])){
+            $output->where('email', $payload['email']);
+        }
+
+        if (!empty($payload['mail_template_ids'])){
+            $output->where('mail_template_id', $payload['mail_template_id']);
+        }
+
+        if (!empty($payload['limit'])){
+            $output->limit($payload['limit'][0], $payload['limit'][1]);
+        }
+
+        $output = Arrays::map($output->order('created_at DESC')->fetchAll(), function (ActiveRow $row){
```

Comment:
Never generate output without knowing what's getting inside - API results should be arrays created explicitly. Set each property of the array yourself and format if necessary.

(e.g. all dates should be returned as RFC3339 dates when possible)

This explicit response definition helps if someone adds new internal column to the table. He/she shouldn't be thinking about which APIs it affects. Also probably not all columns should be available to the API consumers.

Reply:
Regarding date format, RFC3339 is returned by default. But you are right, good point :).

Reply:
It might be RFC3339 now, but up until recently, it used to be Objects with three fields. Let's have the formatting explicit so the reader doesn't have to wonder which way it is and so it's future proof.